### PR TITLE
Update pull_request_template.md

### PR DESCRIPTION
**Pull request by melissaperreault**
**Why are these changes introduced?**

Empower individual contributor to list their testing tasks and help reviewers know what to look for.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -6,6 +6,12 @@ Fixes #0.
 
 **Other considerations**
 
+**Testing**
+- [ ] Mobile
+- [ ] Tablet
+- [ ] Desktop
```
Comment (ludoboludo) [inline .github/PULL_REQUEST_TEMPLATE.md:12]: I almost feel like we just need the one you have after those: 
```
- [ ] _List all the testing tasks that applies to your fix and help peers to review your work._
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -6,6 +6,12 @@ Fixes #0.
...
+- [ ] _List all the testing tasks that applies to your fix and help peers to review your work._
+
```
Comment (martinamarien) [inline .github/PULL_REQUEST_TEMPLATE.md:14]: Love this. Thanks Meli

While we're already updating the template, would you mind including a line under "Demo links" that says something like" 
_Please include a link to a demo store that includes preconfigured sections to allow reviewers to easily test the features you are working on._

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -6,6 +6,12 @@ Fixes #0.
...
 **Demo links**
```
Comment (melissaperreault) [inline .github/PULL_REQUEST_TEMPLATE.md:16]:
```suggestion
_Please include a link to a demo store that includes preconfigured sections and settings to allow reviewers to easily test the features you are working on._
```

Comment (martinamarien) [reply .github/PULL_REQUEST_TEMPLATE.md:17]: I added it!

**Review by martinamarien — APPROVED** [review]
🚢 Thank you Meli 🙏🏻

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -6,7 +6,14 @@ Fixes #0.
 
 **Other considerations**
 
+**Testing steps/scenarios**
+- [ ] Mobile
+- [ ] Tablet
+- [ ] Desktop
```
Comment (melissaperreault) [inline .github/PULL_REQUEST_TEMPLATE.md:12]:
```suggestion
```

**Review by ludoboludo — APPROVED** [review]
🎉

**Pull request merged**